assessments/projects/longhorn/self-assessment.md

```diff
@@ -0,0 +1,436 @@
+
+<hr>
+
+## Security functions and features
+
+Critical Security Components:
+* Volume Encryption: Ensures that data at rest is encrypted, providing confidentiality and protection against unauthorised access. This is a non-configurable element critical for data security, especially when handling sensitive information.
+* Data Replication Integrity: Synchronous replication of data across multiple replicas in different nodes ensures high availability and integrity of data. This is vital for maintaining the state of the data in case of node failure.
+* Automated Snapshot Management: Regular, automatic snapshots protect against data loss by preserving the state of data at specific points in time. These snapshots form the basis for recovery in the event of corruption or data loss incidents.
+
+Security-Relevant Components:
+* Configurable Replication and Backup Settings: While the replication process itself is critical, the ability to configure the number of replicas and backup intervals allows administrators to tailor the redundancy and backup frequency to the risk profile of the data being stored.
+* Kubernetes' Role-Based Access Control (RBAC) Integration: By leveraging Kubernetes RBAC, Longhorn ensures that only authorised users and processes can manage volumes, snapshots, and backups, thus enforcing the principle of least privilege.
+* Health Checks and Monitoring: Longhorn's continuous health checks and monitoring of volumes and replicas are crucial for early detection of potential issues, allowing for proactive remediation before they escalate into security incidents.
+* Backup Data Validation: Before restoring data from backups, Longhorn validates the backup integrity, ensuring that the data has not been tampered with or corrupted, which is essential for the reliability of the restoration process.
+
+
+## Project compliance
+* Compliance: Longhorn adheres to best practices for Kubernetes storage solutions, ensuring it aligns with the security and operational standards expected within the cloud-native ecosystem. However, specific compliance certifications or adherence to standards like PCI-DSS, COBIT, ISO, or GDPR have not been documented.
+
+
+## Secure development practices
+
+Development Pipeline
+* Version Control and Commit Signing: Longhorn uses Git for version control. Contributors are encouraged to sign commits to verify the identity of contributors. More details can be found in the [Getting Started with Longhorn Development](https://github.com/longhorn/longhorn/wiki/Getting-started-with-Longhorn-Development) guide.
+* Code Reviews: Each pull request requires a thorough review by at least two maintainers to ensure quality and security.
```

Comment:
In some of the closed PRs, this doesn't seem to be the case. Please revisit and provide the reference to this process.

Reply:
Thank you for pointing that out, we have fix it.